torch/_export/constraints.py

```diff
@@ -1,48 +1,32 @@
-from typing import Optional, Callable, Union
+import sys
+from typing import Optional
 
 import torch
-from torch import SymInt, SymFloat
-from torch._dynamo import allow_in_graph
-from torch.fx.experimental.symbolic_shapes import constrain_range_int
-from torch.utils._sympy.value_ranges import ValueRangeError
-
-# `Scalar` type used in native_functions.ymal will be translated to `Union[Number, _complex]`
-# could cause type error during since `SymInt` or `SymFloat` will be used.
-# Here manually specify the type explicitly.
-sym_constrain_range: Callable[
-    [Union[int, float, SymInt, SymFloat], Optional[int], Optional[int]],
-    None,
-] = torch.sym_constrain_range  # type: ignore[assignment]
 
 
 # TODO: we want to hide this min/max stuff under some abstraction similar to
 # DynamicDim
-@allow_in_graph
 def constrain_as_value(symbol, min: Optional[int] = None, max: Optional[int] = None):
     """
-    Add min/max constraint on the intermediate symbol at tracing time
+    Add min/max constraint on the intermediate symbol at tracing time. If called in eager mode,
+    it will still check if the input value is within the specified range.
     """
 
-    if not isinstance(symbol, SymInt):
-        constrain_range_int(symbol, min=min, max=max)
-    else:
-        sym_constrain_range(symbol, min, max)
+    if min is None:
+        min = 0
+    if max is None:
+        max = sys.maxsize
```

Comment:
I still don't think you should do this. What's the point of the op taking optional arguments if you always fill them in?